File: src/main/scala/qmf/poc/agent/catalog/models/ObjectRemarks.scala
package qmf.poc.agent.catalog.models

import spray.json.{JsObject, JsString, JsValue, JsonFormat}

class ObjectRemarks(
    val owner: String,
    val name: String,
    val `type`: String,
    val remarks: String
)

object ObjectRemarks:
  given JsonFormat[ObjectRemarks] with
    def write(o: ObjectRemarks): JsObject =
      JsObject(
        "name" -> JsString(o.name),
        "owner" -> JsString(o.owner),
        "type" -> JsString(o.`type`),
        "remarks" -> JsString(o.remarks)
      )

    //noinspection NotImplementedCode
    def read(value: JsValue): ObjectRemarks = ???

File: src/main/java/qmf/poc/agent/catalog/models/ObjectRemarks.java
package qmf.poc.agent.catalog.models;

public class ObjectRemarks {
    final String owner;
    final String name;
    final String type;
    final String remarks;

    public ObjectRemarks(String owner, String name, String type, String remarks) {
        this.owner = owner;
        this.name = name;
        this.type = type;
        this.remarks = remarks;
    }

    @Override
    public String toString() {
        String s = "'" + owner + "'|'" + name + "'|'" + type + "'|'" + remarks;// + "'";
        return s.length() <= 131 ? s + "'" : s.substring(0, 129) + "...'";
    }
}